File: swift/triangle/Sources/Triangle/Triangle.swift
class Triangle {

    public var kind: String = ""

    public init(_ i: Double, _ i2: Double, _ i3: Double) {
        let doubles = [i, i2, i3].sorted()

        doubles.forEach {
            if $0 <= 0 {
                kind = "ErrorKind"
            }
        }

        if doubles[0] + doubles[1] < doubles[2] {
            kind = "ErrorKind"
        }

        if !kind.isEmpty {
            return
        }

        let s = Set(doubles)

        if s.count == 1 {
            kind = "Equilateral"
        } else if s.count == 3 {
            kind = "Scalene"
        } else {
            kind = "Isosceles"
        }

    }
}

File: prolog/triangle/triangle.pl
triangle(A, B, C) :- A > 0, B > 0, C > 0, A + B >= C, B + C >= A, C + A >= B.
triangle(A, A, A, "equilateral") :- triangle(A, A, A), !.
triangle(A, A, B, "isosceles") :- triangle(A, A, B), !.
triangle(A, B, A, "isosceles") :- triangle(A, B, A), !.
triangle(B, A, A, "isosceles") :- triangle(B, A, A), !.
triangle(A, B, C, "scalene") :- triangle(A, B, C), A\==B, B\==C, C\==A.
